src/Microsoft.SqlTools.ServiceLayer/Connection/Contracts/ConnectionDetails.cs

```diff
@@ -592,31 +592,41 @@ public string AzureAccountToken
         }
 
         public bool IsComparableTo(ConnectionDetails other)
-        {
-            if (other == null)
-            {
-                return false;
-            }
-
-            if (ServerName != other.ServerName
-                || AuthenticationType != other.AuthenticationType
-                || UserName != other.UserName
-                || AzureAccountToken != other.AzureAccountToken)
-            {
-                return false;
-            }
-
-            // For database name, only compare if neither is empty. This is important
-            // Since it allows for handling of connections to the default database, but is
-            // not a 100% accurate heuristic.
-            if (!string.IsNullOrEmpty(DatabaseName)
-                && !string.IsNullOrEmpty(other.DatabaseName)
-                && DatabaseName != other.DatabaseName)
-            {
-                return false;
-            }
-
-            return true;
-        }
+            => other != null
```

Comment:
I'm a little concerned if we have a different "connection equality" check in STS than in ADS that we can get weird side-effects in various scenarios when editing the connections in ADS.  If you've done sufficient testing across these kinds of end-to-end scenarios then this is fine.

Reply:
My main concern here is that if STS is making changes to the connection details itself that could result in unexpected disconnects/reconnects since it would have different properties than what ADS has which could break customers if they're using temp objects or other transient functionality. 

It would be great to have a better sense of how/when changes to the connection details could happen (both in ADS and STS) to make sure we aren't going to be regressing any normal behavior here.

Reply:
> I'm a little concerned if we have a different "connection equality" check in STS than in ADS that we can get weird side-effects in various scenarios when editing the connections in ADS.

That shouldn't be of concern, as ADS seems to delegate all settings comparisons to STS layer and only works with Connection Profile (that is identity) with whatever settings it's provided with. STS is responsible to cache identity and settings combo in `OwnerToConnectionMap` and that's the heart of all saved settings that apply to connection profiles. ADS would always show new properties on modifying connection settings even if STS did not update the settings in the map so that was a big design problem and affected reliability of ADS connections. ADS is always doing the right thing.
 
> My main concern here is that if STS is making changes to the connection details itself that could result in unexpected disconnects/reconnects since it would have different properties than what ADS has which could break customers if they're using temp objects or other transient functionality.

There won't be any unexpected disconnects, as the connections will only be updated with new settings on a successful save and will only affect new connections created from that profile. And that's exactly what is needed to be able to edit a connection. We should be forcing a reconnect with new advanced options before saving it, and on saving, new properties must apply on everything new that starts with this edited connection.